extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/KafkaConnectSources.java

```diff
@@ -71,13 +79,33 @@ public static StreamSource<SourceRecord> connect(@Nonnull Properties properties)
         //fail fast, required by lazy-initialized KafkaConnectSource
         checkRequiredProperty(properties, "connector.class");
 
-        return SourceBuilder.timestampedStream(name, ctx ->
-                        new KafkaConnectSource(properties))
-                .fillBufferFn(KafkaConnectSource::fillBuffer)
-                .createSnapshotFn(KafkaConnectSource::createSnapshot)
-                .restoreSnapshotFn(KafkaConnectSource::restoreSnapshot)
-                .destroyFn(KafkaConnectSource::destroy)
-                .build();
+        int tasksMax = Integer.parseInt(properties.getProperty("tasks.max", "1"));
+
+        return Sources.streamFromProcessorWithWatermarks(name, true,
+                eventTimePolicy -> ProcessorMetaSupplier.preferLocalParallelismOnSingleMember(
+                        new ProcessorSupplier() {
+                            private ConnectorWrapper connectorWrapper;
```

Comment:
not necessary, but adding `transient` makes it clear that it is not serialized

Reply:
fixed: https://github.com/hazelcast/hazelcast/pull/23831/commits/dc7f65e44e6beae45281092eba18cc60a51ac754